### PR TITLE
IDC-1532 multiple series search google

### PR DESCRIPTION
**Pull request by JamesAPetts**
RE: #1532 

Google Healthcare (and potentialy other platforms), only supports a single query parameter, so you can't filter on multiple `SeriesInstanceUIDs`.

This PR splits up the calls if using the google cloud adapter (I.e. if `enableGoogleCloudAdapter` is `true` in your app config).

### CODE REVIEW

File: platform/core/src/studies/retrieveStudiesMetadata.js [hunk 1]
```diff
@@ -11,20 +11,28 @@ import { retrieveStudyMetadata } from './retrieveStudyMetadata';
  * @param {Array} studyInstanceUIDs The UIDs of the Studies to be retrieved
  * @param {Object} [filters] - Object containing filters to be applied on retrieve metadata process
  * @param {string} [filter.seriesInstanceUID] - series instance uid to filter results against
+ * @param {boolean} [seperateSeriesInstanceUIDFilters = false] - If true, split filtered metadata calls into multiple calls,
```
Comment (swederik) [inline platform/core/src/studies/retrieveStudiesMetadata.js:14]: Typo in separate

File: platform/core/src/studies/retrieveStudyMetadata.js [hunk 2]
```diff
@@ -33,18 +40,75 @@ export function retrieveStudyMetadata(server, StudyInstanceUID, filters) {
...
+    filters.seriesInstanceUID &&
+    seperateSeriesInstanceUIDFilters
+  ) {
+    promise = _seperateSeriesRequestToAggregatePromise(
+      server,
+      StudyInstanceUID,
+      filters
+    );
+  } else {
+    promise = new Promise((resolve, reject) => {
```
Comment (swederik) [inline platform/core/src/studies/retrieveStudyMetadata.js:56]: what is the point of this new promise? isn't it the same as promise = RetrieveMetadata(server, StudyInstanceUID, filters)?

File: platform/core/src/studies/retrieveStudyMetadata.js [hunk 3]
```diff
@@ -33,18 +40,75 @@ export function retrieveStudyMetadata(server, StudyInstanceUID, filters) {
...
+  }
 
   // Store the promise in cache
   StudyMetaDataPromises.set(StudyInstanceUID, promise);
 
   return promise;
 }
 
+/**
+ * Splits up seriesInstanceUID filters to multiple calls for platforms
```
Comment (swederik) [inline platform/core/src/studies/retrieveStudyMetadata.js:70]: can we capitalize SeriesInstanceUID?

File: platform/viewer/public/config/idc.js [hunk 4]
```diff
@@ -7,9 +7,12 @@ window.config = function(props) {
     enableGoogleCloudAdapter: true,
     enableGoogleCloudAdapterUI: false,
     showStudyList: true,
+    enableStudyLazyLoad: false,
```
Comment (swederik) [inline platform/viewer/public/config/idc.js:10]: Why change this file? and why false?

File: platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js [hunk 5]
```diff
@@ -340,6 +341,10 @@ function ViewerRetrieveStudyData({
         }
       }
 
+      if (appConfig.enableGoogleCloudAdapter) {
+        retrieveParams.push(true); // Seperate SeriesInstanceUID filter calls.
```
Comment (swederik) [inline platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js:345]: This is really weird. Let's hope we never reorganize the arguments to the function.

**Review by swederik — APPROVED** [review]
Seems fine if you fix the spelling of separate

File: platform/core/src/studies/retrieveStudyMetadata.js (hunk 3 above)
Comment (JamesAPetts) [reply platform/core/src/studies/retrieveStudyMetadata.js:70]: I'm on the fence about this, `seriesInstanceUID` is the actual parameter passed to dicomweb client, that is the parameter this splits up. So I think it makes sense to have the jsdoc reflect this.

File: platform/viewer/public/config/idc.js (hunk 4 above)
Comment (JamesAPetts) [reply platform/viewer/public/config/idc.js:10]: reverted.

File: platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js (hunk 5 above)
Comment (JamesAPetts) [reply platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js:345]: I agree, I'd rather have a deconstructed object inside `retrieveStudiesMetadata`, but it'd change what is actually top level API in `@ohif/core`.

File: platform/core/src/studies/retrieveStudyMetadata.js (hunk 2 above)
Comment (JamesAPetts) [reply platform/core/src/studies/retrieveStudyMetadata.js:56]: This is just what was in the previous code. Hmm you are right, weird. Removed and tested, all works fine.

**Pull request merged**